gencrud/configuraton.py

```diff
@@ -100,43 +102,57 @@ def __init__( self, filename = None, **cfg ) -> None:
         else:
             self.__config       = cfg
 
-        self.__config.update( self.__config.get( 'project_defaults', {} ) )
+        # TODO: Veryfy the loaded template against the schema
+        try:
+            jsonschema.Draft7Validator(GENCRUD_SCHEME)
+            jsonschema.validate(instance=self.__config, schema=GENCRUD_SCHEME)
+
+        except jsonschema.SchemaError as exc:
+            print(exc)
+            raise SystemExit
+
+        except jsonschema.ValidationError as exc:
+            print(exc)
+            raise SystemExit
+
         self.__controls     = None
-        if C_OBJECTS not in self.__config:
-            raise MissingAttribute( C_ROOT, C_OBJECTS )
+        if 'objects' not in self.__config:
+            raise MissingAttribute( 'root', 'objects' )
 
-        if C_APPLICATION not in self.__config:
-            raise MissingAttribute( C_ROOT, C_APPLICATION )
+        if 'application' not in self.__config:
+            raise MissingAttribute( 'root', 'application' )
 
-        controls            = cfg.get( C_CONTROLS, None )
+        controls            = cfg.get( 'controls', None )
         if controls is not None:
             self.__controls = DymanicControls( controls )
 
-        self.__options      = TemplateOptions( self.__config )
-        self.__source       = SourceLocation( self,  self.__config )
-        self.__template     = TemplateLocation( self, self.__config )
-        self.__references   = TemplateReferences( self.__config )
+        opts                = self.__config[ C_OPTIONS ] if C_OPTIONS in self.__config else { }
+        self.__options      = TemplateOptions( **opts )
+        self.__python       = TemplateSourcePython( **self.__config )
```

Comment:
I think this will crash in the open source version.  The TemplateLocation class from location.py should be used instead of the TemplateSourcePython from source.py. The reason is because the information where the source and where the templates are need to be separated from each other. If a user does not specify anything, the gencrud default templates should be taken. This, however, should not hold for the output folder. The output folder, i.e. the source folder, must be strictly specified